Purchases/Public/Purchases+async.swift

```diff
@@ -0,0 +1,271 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  Purchases+async.swift
+//
+//  Created by Andrés Boedo on 24/11/21.
+
+import Foundation
+import StoreKit
+
+/// This extension holds the biolerplate logic to convert methods with completion blocks into async / await syntax.
+extension Purchases {
+
+    @available(iOS 15.0, macOS 12, tvOS 15.0, watchOS 8.0, *)
+    func logInAsync(_ appUserID: String) async throws -> (CustomerInfo, Bool) {
+        return try await withCheckedThrowingContinuation { continuation in
+            logIn(appUserID) { maybeCustomerInfo, created, maybeError in
+                if let error = maybeError {
+                    continuation.resume(throwing: error)
+                    return
+                }
+                guard let customerInfo = maybeCustomerInfo else {
```

Comment:
This to me is the best reason to use the `async` variants in Swift. Since we can't use `Result<(CustomerInfo, Bool), Error>` for Objective-C compatibility, at least this API now eliminates the 2 invalid cases: `(nil, nil)`, `(.some, .some)`.

Reply:
Agreed. I'd even been considering making some of our methods with nullable value + nullable error unavailable for swift, and adding a Result variant, but now that Swift concurrency got backported to iOS 13+, I think most devs will be able to just use this version instead and be guaranteed a correct value